src/transformers/models/rwkv/tokenization_rwkv.py

```diff
@@ -0,0 +1,525 @@
+        l = len(content)
+        result = [
+            [
+                "",
+            ],
+        ]
+        while node is not None and idx < l:
+            w = content[idx]
+            if not node.has_next(w):
+                return result[-1]
+                # last word
+            node = node.get_node(w)
+            if node.is_leaf():
+                result.append([content[start_idx : idx + 1], node.get_data()])
+            idx += 1
+        return result[-1]
+
+    def max_score(self, content, start_idx):
+        idx = start_idx
+        node = self.root
+        l = len(content)
+        result = [
+            ["", (3, 0)],
+        ]
+        while node is not None and idx < l:
+            w = content[idx]
+            if not node.has_next(w):
+                break
+                # last word
+            node = node.get_node(w)
+            if node.is_leaf():
+                result.append([content[start_idx : idx + 1], node.get_data()])
+            idx += 1
+        if len(result) > 1:
+            result = sorted(result, key=lambda x: x[1][1])
+        return result[-1]
+
+    def match(self, content, add_unk=True, unk_id=-1, **kwargs):
+        # length
+        l = len(content)
+        i = 0
+        result_list = []
+        while i < l:
+            match_word = self.max_prefix(content=content, start_idx=i)
+            # print(match_word)
+            w = match_word[0]
+            if len(w) > 0:
+                result_list.append(match_word[1])
+                i += len(w)
+            else:
+                if add_unk:
+                    result_list.append(unk_id)
+                i += 1
+        return result_list
+
+    def id2str(self, ids, escape_special_ids=True, end_ids=[], **kwargs):
+        res_str = ""
+        for rid in ids:
+            if rid in self.r_data:
+                if rid in end_ids:
+                    break
+                if escape_special_ids and rid in self.special_ids:
+                    continue
+                rstr = self.r_data[rid]
+                res_str += rstr
+            elif rid == 0:
+                break
+            else:
+                print("ERROR unknown id %d" % rid)
+                res_str += "UNK"
+        return res_str
+
+    def id2str_v2(self, ids, escape_special_ids=True, end_ids=[], **kwargs):
+        res_str = ""
+        for rid in ids:
+            if rid in self.r_data:
+                if rid in end_ids:
+                    break
+                rstr = self.r_data[rid]
+                if escape_special_ids and rid in self.special_ids:
+                    continue
+                res_str += rstr
+            elif rid == 0:
+                break
+            else:
+                print("ERROR unknown id %d" % rid)
+                res_str += "UNK"
+        return res_str
+
+
+class RWKVWorldTokenizer(PreTrainedTokenizer):
+    """
+    Construct a RWKVWorldTokenizer. Based on byte-level Byte-Pair-Encoding.
```

Comment:
We have a lot of BPE based tokenizers like GPT2 for example, would it work to use them?

Reply:
For the RWKV World Tokenizer, it terminates at the inferred token id=0 (https://github.com/huggingface/transformers/pull/26963/files#diff-6b1cfe4e721b792b5bdfde6f7a2645a9a92fcf7b9e3fb0e8c81f040759a7b6d0R204), and there is no explicit eos (end of sentence) token in the vocabulary. This is the main source of difference.

Reply:
Then we should just manually set the token 0 to be the eos token. The content does not really matter you can choose whatever no?